package/MDAnalysis/analysis/bat.py

```diff
@@ -0,0 +1,503 @@
+
+        if bat_nc is not None:
+            self.load_bat(bat_nc)
+
+    def _prepare(self):
+        self.bat = []
+
+    def _single_frame(self):
+        # Calculate coordinates based on the root atoms
+        # The rotation axis is a normalized vector pointing from atom 0 to 1
+        # It is described in two degrees of freedom
+        # by the polar angle and azimuth
+        if (self._root.dimensions[:3] == 0).any():
+            (p0, p1, p2) = self._root.positions
+        else:
+            (p0, p1, p2) = make_whole(self._root, inplace=False)
+        v01 = p1 - p0
+        v21 = p1 - p2
+        # Internal coordinates
+        r01 = np.sqrt(np.sum(v01 *
+                             v01))  # Distance between first two root atoms
+        r12 = np.sqrt(np.sum(v21 *
+                             v21))  # Distance between second two root atoms
+        a012 = np.arccos(max(-1.,min(1.,np.sum(v01*v21)/\
+          np.sqrt(np.sum(v01*v01)*np.sum(v21*v21))))) # Angle between root atoms
+        # Exernal coordinates
+        e = v01 / r01
+        phi = np.arctan2(e[1], e[0])  # Polar angle
+        theta = np.arccos(e[2])  # Azimuthal angle
+        # Rotation to the z axis
+        cp = np.cos(phi)
+        sp = np.sin(phi)
+        ct = np.cos(theta)
+        st = np.sin(theta)
+        Rz = np.array([[cp * ct, ct * sp, -st], [-sp, cp, 0],
+                       [cp * st, sp * st, ct]])
+        pos2 = Rz.dot(p2 - p1)
+        # Angle about the rotation axis
+        omega = np.arctan2(pos2[1], pos2[0])
+        root_based = np.concatenate((p0, [phi, theta, omega, r01, r12, a012]))
+
+        # Calculate internal coordinates from the torsion list
+        bonds = calc_bonds(self._ag1.positions,
+                           self._ag2.positions,
+                           box=self._ag1.dimensions)
+        angles = calc_angles(self._ag1.positions,
+                             self._ag2.positions,
+                             self._ag3.positions,
+                             box=self._ag1.dimensions)
+        torsions = calc_dihedrals(self._ag1.positions,
+                                  self._ag2.positions,
+                                  self._ag3.positions,
+                                  self._ag4.positions,
+                                  box=self._ag1.dimensions)
+        # When appropriate, calculate improper torsions
+        shift = torsions[self._primary_torsion_indices]
+        shift[self._unique_primary_torsion_indices] = 0.
+        torsions -= shift
+        # Wrap torsions to between -np.pi and np.pi
+        torsions = ((torsions + np.pi) % (2 * np.pi)) - np.pi
+
+        self.bat.append(np.concatenate((root_based, bonds, angles, torsions)))
+
+    def load_bat(self, FN):
+        """Loads the bat trajectory from a netcdf file
+        """
+        nc_F = Dataset(FN,'r')
+        self.bat = list(np.array(nc_F.variables['bat']))
+        nc_F.close()
```

Comment:
If you cast `bat` to array as suggested above then you don't need the cast here.
```python
        with Dataset(filename,'r') as nc_F:
            self.bat = np.asarray(nc_F.variables['bat'])
```
(`asarray()` will only do the cast if necessary... I think ncdf already gives you an array, right?)

Reply:
I'm switching to numpy binaries because the API is simpler and there are fewer compatibility issues. The code was not passing continuous integration tests with netCDF4.